### ISSUE TITLE
Java role has (oracle)JDK v7 hardcoded

### ISSUE THREAD

**Issue by lhoss** [cloudera/cloudera-playbook#24]
I just hope nobody used it and excluded it in the site.yml playbook (or by using tags, but not 'java' tag) ..
since JDK8 is more than recommended since some years now

As it's not the idea to use external ansible roles here (of course a large choice out there for Java/JDK installs), 2 good solutions:
* (1) re-use (copy paste in this case) the flexible java deployment, supporting both openjdk/oraclejdk from:
https://github.com/hortonworks/ansible-hortonworks/blob/master/playbooks/roles/common/tasks/java.yml  
  * and remove any distro specific logic (in this case the suse tasks)
* (2) or simply just yum install java-1.8.0-openjdk-devel (recent versions have strong crypto builtin, so no more need for `JCE unlimited policy` tasks!)

**Comment by lhoss** [comment]
Already done locally a quick version of (2), can contribute via PR, if anyone interested

git diff is
```
iff --git a/roles/java/defaults/main.yml b/roles/java/defaults/main.yml
new file mode 100644
index 0000000..b6238fb
--- /dev/null
+++ b/roles/java/defaults/main.yml
@@ -0,0 +1,3 @@
+# You can disable java installation by simply setting the java var to empty string
+java: "openjdk"
+openjdk_package: java-1.8.0-openjdk-devel
diff --git a/roles/java/tasks/main.yml b/roles/java/tasks/main.yml
index c08ad99..8388247 100644
--- a/roles/java/tasks/main.yml
+++ b/roles/java/tasks/main.yml
@@ -1,46 +1,9 @@
 ---
-
-- name: Install Oracle JDK
-  yum: name={{ item }} state=latest update_cache=yes
-  with_items:
-    - oracle-j2sdk1.7
-    - unzip
-
-- stat: path="{{ tmp_dir }}/UnlimitedJCEPolicyJDK7.zip"
-  register: jce_zip_exists
-
-- name: Download JCE unlimited policy
-  get_url:
-    url=http://download.oracle.com/otn-pub/java/jce/7/UnlimitedJCEPolicyJDK7.zip
-    dest="{{ tmp_dir }}/UnlimitedJCEPolicyJDK7.zip"
-    headers="Cookie:oraclelicense=accept-securebackup-cookie"
-  when: jce_zip_exists.stat.exists == False
-
-- name: Unzip JCE unlimited policy files
-  unarchive:
-    src: "{{ tmp_dir }}/UnlimitedJCEPolicyJDK7.zip"
-    dest: "{{ tmp_dir }}"
-    copy: no
-
-- name: Install local_policy.jar
-  copy:
-    src: "{{ tmp_dir }}/UnlimitedJCEPolicy/local_policy.jar"
-    dest: /usr/java/jdk1.7.0_67-cloudera/jre/lib/security/local_policy.jar
-    backup: yes
-    remote_src: True
-
-- name: Install US_export_policy.jar
-  copy:
-    src: "{{ tmp_dir }}/UnlimitedJCEPolicy/US_export_policy.jar"
-    dest: /usr/java/jdk1.7.0_67-cloudera/jre/lib/security/US_export_policy.jar
-    backup: yes
-    remote_src: True
-
-- name: Cleanup tmp files
-  file:
-    path="{{ tmp_dir }}/{{ item }}"
-    state=absent
-  with_items:
-    - UnlimitedJCEPolicy
-    - UnlimitedJCEPolicyJDK7.zip
-  ignore_errors: True
\ No newline at end of file
+- name: Install OpenJDK
+  package:
+    name: "{{ openjdk_package }}"
+    update_cache: yes
+    state: present
+  when: java == "openjdk"
+
+# Optional: Add support for Oracle JDK, see https://github.com/hortonworks/ansible-hortonworks/blob/master/playbooks/roles/common/tasks/java.yml

```

**Comment by roczei** [comment]
Hi @lhoss ,

You need to first sign and return an ICLA and CCLA before we can accept and redistribute your contribution. Once these are submitted you are free to start contributing to cloudera-playbook. Submit these to CLA@cloudera.com

Here you can find the pdf files:

https://github.com/cloudera/cloudera-playbook/tree/master/icla

Please update this issue (24) if you receive the confirmation from our CLA team.

Thank you!

**Comment by lhoss** [comment]
> You need to first sign and return an ICLA and CCLA before we can accept and redistribute your contribution. Once these are submitted you are free to start contributing to cloudera-playbook. Submit these to [CLA@cloudera.com](mailto:CLA@cloudera.com)

Hi @roczei 
Yes I plan to take care of those Docs latest end of this week, and I assume no legal Blockers
so you could already give your opinion about my PRs
( Even easier would be to have a look at our fork and there the `scigility` branch, which simply contains *all* my submitted PRs including solved conflicts between the atomic PRs)

**Comment by dbeech** [comment]
Fixed by #38
